model/actor.gd

```diff
@@ -71,6 +74,18 @@ func step_time():
 	if can_draw():
 		draw_cooldown -= draw_rate
 
+func set_upgrade(upgrade):
+	if upgrades.size() == upgrade_slot:
+		upgrades.pop_front();
```

Comment:
The expected behavior from the design document was to not allow new upgrades if there are no slots available.